apps/web/src/app/core/router.service.ts

```diff
@@ -51,11 +54,50 @@ export class RouterService {
       });
   }
 
-  getPreviousUrl() {
+  getPreviousUrl(): string {
```

Comment:
These can return undefined, so I think a more accurate type is to include undefined.

```suggestion
  getPreviousUrl(): string | undefined {
```